clang/lib/3C/ConstraintBuilder.cpp

```diff
@@ -398,6 +399,47 @@ class FunctionVisitor : public RecursiveASTVisitor<FunctionVisitor> {
   TypeVarInfo &TVInfo;
 };
 
+class PtrToStructDef : public RecursiveASTVisitor<PtrToStructDef> {
```

Comment:
Add comment above this, saying what it is for and a sentence about where it's used.

Also, will this do the right thing for function pointers? E.g.,
```
struct foo { int *p; };
typedef struct foo (*foofp)(int *q);
void bar(void) {
  foofp *s = 0;
}
```
The function pointer returns a `struct foo`, which is not a pointer, but also has a pointer in its argument. Will your code get confused and think this typedef contains a pointer to a struct?

Reply:
Ah I have probably gotten this wrong: You are looking for a struct *definition* in the typedef, not a reference to it. In that case, maybe a variation of my example?
```
typedef struct { int *p; } (*foofp)(int *q);
void bar(void) {
  foofp *s = 0;
}

Reply:
Yep! That got it!
`_Ptr<struct (anonymous struct at /home/aaron/build/bin/test.c:1:9) (_Ptr<int> )> s = 0;`